### PR TITLE
Diona hydration can prevent ignition by damage

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Diona have a significant downside of lighting on fire when faced with any laser weapons. This makes them one of the weakest species in the game (besides the lack of shoes).

This PR reduces that downside a bit by making it so that heat damage will reduce their thirst level, and if their thirst ever reaches zero, they will ignite on any subsequent damage they've taken.

## Why / Balance

Being a Diona is suffering

## Technical details

My comments are quite straightforward as well as the code

## Media

This video is a showcase what a 4x dehydration multiplier looks like. The PR has it set to 2x.

https://private-user-images.githubusercontent.com/58439124/335835870-82ff06ac-43ac-4460-8cfe-7ae41101844f.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3Njk5MjIsIm5iZiI6MTcyMzc2OTYyMiwicGF0aCI6Ii81ODQzOTEyNC8zMzU4MzU4NzAtODJmZjA2YWMtNDNhYy00NDYwLThjZmUtN2FlNDExMDE4NDRmLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE2VDAwNTM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYzkyYzlhZTQyYjMyZGJlMGQ1YzAzMDFkZmUzYTlkYzZiZDY4NDE3ZjA3ZGIxZDQ1ODlmN2FhMTRjMWU1NjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WTlCpWHWg7EPzl-nk5_L_M8UvwxAbMZupPNipHEap7w

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Heat damage will cause Diona to lose hydration. Upon max dehydration, Diona will ignite on damage.

### CODE REVIEW

**Comment by Emisse** [comment]
im not sure if this should be tied into flammable system directly but might want a second opinion

**Comment by TheShuEd** [comment]
yes, it should be some separate system and component

File: Content.Server/Atmos/EntitySystems/FlammableSystem.cs [hunk 1]
```diff
@@ -351,11 +353,26 @@ private void OnDamageChanged(EntityUid uid, IgniteOnHeatDamageComponent componen
...
+            if (args.DamageDelta.DamageDict.TryGetValue("Heat", out FixedPoint2 heatDamage))
             {
+                // Ignition can be prevented if the PreventedByHydration is true
+                // Reduce the user's hydration level by the damage times the mulitiplier
+                if (component.PreventedByHydration)
+                {
+                    if (TryComp<ThirstComponent>(uid, out var thirst))
+                    {
+                        // Thirst needs to be a negative value to go down
+                        EntityManager.System<ThirstSystem>().ModifyThirst(uid, thirst, heatDamage.Float() * -component.DehydrationMultiplier);
```
Comment (doublechest0) [inline Content.Server/Atmos/EntitySystems/FlammableSystem.cs:366]: You should add `[Dependency] private readonly ThirstSystem _thirst = default!;`
```suggestion
                        _thirst.ModifyThirst(uid, thirst, heatDamage.Float() * -component.DehydrationMultiplier);
```

**Comment by DieselMohawk** [comment]
Heya, will you get back to this sometime?

**Comment by Plykiya** [comment]
> Heya, will you get back to this sometime?

sure, I'm in a PR mood lately
give me a few hours

**Comment by DieselMohawk** [comment]
> sure, I'm in a PR mood lately give me a few hours

Awesome, thank you.

**Pull request closed without merging**